FinalTask/app.py

```diff
@@ -0,0 +1,202 @@
+from fastapi import FastAPI, HTTPException, Depends
+from sqlalchemy import create_engine, Column, Integer, String, Float, ForeignKey, DateTime
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker, relationship, Session
+from pydantic import BaseModel
+from typing import List
+from datetime import datetime
+
+DATABASE_URL = "sqlite:///./final.db"
+Base = declarative_base()
+
+engine = create_engine(DATABASE_URL, connect_args={"check_same_thread": False})
+SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+
+class User(Base):
+    __tablename__ = "users"
+    id = Column(Integer, primary_key=True, index=True)
+    name = Column(String)
+    surname = Column(String)
+    email = Column(String, unique=True, index=True)
+    password = Column(String)
+
+class Product(Base):
+    __tablename__ = "products"
+    id = Column(Integer, primary_key=True, index=True)
+    name = Column(String)
+    description = Column(String)
+    price = Column(Float)
+
+class Order(Base):
+    __tablename__ = "orders"
+    id = Column(Integer, primary_key=True, index=True)
+    user_id = Column(Integer, ForeignKey("users.id"))
+    product_id = Column(Integer, ForeignKey("products.id"))
+    order_date = Column(DateTime)
+    status = Column(String)
+    user = relationship("User")
+    product = relationship("Product")
+
+Base.metadata.create_all(bind=engine)
+
+class UserCreateModel(BaseModel):
+    name: str
+    surname: str
+    email: str
+    password: str
+
+class UserReadModel(BaseModel):
+    id: int
+    name: str
+    surname: str
+    email: str
+
+class ProductCreateModel(BaseModel):
+    name: str
+    description: str
+    price: float
+
+class ProductReadModel(BaseModel):
+    id: int
+    name: str
+    description: str
+    price: float
```

Comment:
Модели очень похожи, а значит можно создать одну из них, наследуясь от другой.
```suggestion
class ProductReadModel(ProductCreateModel):
    id: int
```